cal-itp/benefits
# Redesign: Sign in with Login.gov button + Eligibility Start page alignment

#465 

## What this PR does
- Adds `Login.gov` SVG file, from [18F's Identity Style Guide docs](https://github.com/18F/identity-style-guide): https://design.login.gov/brand/#logo
- Updates both English and Spanish label copy (final translations needed) for the button to `Create an account or sign in with`
- Adds a 60px high button that reads has a `Login.gov` SVG, also in `#ffffff`. Button background is `#046b99` and changes to `#2f4c65` on hover. Within the `<a>` link in the DOM, there is text that can be read by screenreaders, but is not visible in the browser:
<img width="583" alt="image" src="https://user-images.githubusercontent.com/3673236/165400519-47d63a72-abe1-445c-bc67-53428cec4327.png">


## Code changes
- Removes the previous `Button#external` view model class/method and replaces it with a a Login.gov-specific view model method. This new method, `Button#login` should only be used to create a Login.gov button.
- By default, the `Button#login` method adds the following:

1. `#login` ID, which adds the Login.gov SVG
2. `Login.gov` fallback text (This fallback text is the same for both English and Spanish languages.)
- The `fallback_text` attribute on Button can be used separately from this `login` button method as well, by others in the future.

## How to test this PR
- Developers: Make sure to run `bin/init.sh` before you test this locally (or else the button will read `Continue with login.gov login.gov`
- Test that this button renders as it should on Desktop, Mobile
- Test that you can hover over this button with your mouse to get the hover effect
- Test that you can focus onto this button with keyboard only
- Ensure no regressions on DEFTL flow without Login.gov

## Button design & composition

Button
<img width="557" alt="image" src="https://user-images.githubusercontent.com/3673236/165400578-32539a54-463c-4f4f-8502-1332763fabb1.png">

Hover
<img width="556" alt="image" src="https://user-images.githubusercontent.com/3673236/165400605-ba848c5b-9bc9-4fbe-b640-ac536b5ba15e.png">



On the page, in various mobile widths

| iPhone 5         | iPhone 6     | iPhone SE | iPhone XS, 12 Mini|
|--------------|-----------|------------|------------|
|<img width="456" alt="image" src="https://user-images.githubusercontent.com/3673236/165400888-782c23fe-fd8f-4e1c-845e-c043bae35a19.png">|<img width="448" alt="image" src="https://user-images.githubusercontent.com/3673236/165400921-9b3c53ae-110c-4687-ade6-9ae8a6ccc13c.png">|<img width="453" alt="image" src="https://user-images.githubusercontent.com/3673236/165400964-bb9f36d9-8261-4eed-9e59-8340bb660f12.png">|<img width="483" alt="image" src="https://user-images.githubusercontent.com/3673236/165400983-8d7ecd56-4046-4852-8541-da70b34cd397.png">        |



<img width="1512" alt="image" src="https://user-images.githubusercontent.com/3673236/165557117-1d759b17-96c4-4ce9-8228-10a7c365cea1.png">
<img width="702" alt="image" src="https://user-images.githubusercontent.com/3673236/165557258-873e0b8d-30d4-49f7-bed4-d72c7ce43fd7.png">
<img width="557" alt="image" src="https://user-images.githubusercontent.com/3673236/165557314-36f2b89a-6598-46b8-b88b-8ce76fccbeab.png">
